recipes/nginx.rb

```diff
@@ -0,0 +1,35 @@
+#
+# Cookbook Name:: wordpress
+# Recipe:: nginx 
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+yum_package 'nginx' do
+    action :install
+end
+
+template "/etc/nginx/nginx.conf" do
+  source 'nginx.conf.erb'
+  variables(
+    :upstream1          => node['wordpress']['app1']['host'],
+    :upstream2          => node['wordpress']['app2']['host']
+  )
+  action :create
+end
```

Comment:
I guess when this solution is deployed in a chef environment (chef server and clients), the chef-client will run periodically and check if nginx configuration is up to date with the recipe. Am I right?